vyper/ast/nodes.py

```diff
@@ -1290,6 +1290,18 @@ def __init__(self, *args, **kwargs):
             elif call_name == "public":
                 # declaring a public variable
                 self.is_public = True
+                # handle the cases where a constant or immutable variable is public:
+                mutability = next(
+                    (
+                        i.func.id
+                        for i in self.annotation.get_children(
+                        Call, filters={"func.id": {"immutable", "constant"}}
```

Comment:
this is weird. you might be able to have some weird combination like `public(immutable(constant(...)))`